langchain/chat_models/google_palm.py

```diff
@@ -0,0 +1,253 @@
+"""Wrapper around Google's PaLM Chat API."""
+from typing import Any, Dict, List, Optional
+
+from pydantic import BaseModel, root_validator
+
+from langchain.chat_models.base import BaseChatModel
+from langchain.schema import (
+    AIMessage,
+    BaseMessage,
+    ChatGeneration,
+    ChatMessage,
+    ChatResult,
+    HumanMessage,
+    SystemMessage,
+)
+from langchain.utils import get_from_dict_or_env
+
+try:
+    # Import only for type checking.
+    import google.generativeai as genai
+except ImportError:
+    pass
```

Comment:
for type checking you can do something like
```python
from typing import TYPE_CHECKING

if TYPE_CHECKING:
  import google.generativeai as genie
```

Reply:
if you add `from __future__ import annotations` should work perfectly and remove need for quotes, etc

Reply:
Adding `from __future__ import annotations` let me clean up some quotes and imports further down in the file. I still needed the TYPE_CHECKING to stop import failing if the user hasn't installed the genai Python package.